src/hooks/v1/RelaunchV1ViaV3Create.ts

```diff
@@ -0,0 +1,148 @@
+import { SECONDS_IN_DAY } from 'constants/numbers'
+import { ProjectMetadataContext } from 'contexts/projectMetadataContext'
+import { V1ProjectContext } from 'contexts/v1/projectContext'
+import { PayoutMod, TicketMod } from 'models/mods'
+import { consolidateMetadata } from 'models/project-metadata'
+import { useRouter } from 'next/router'
+import { useCallback, useContext, useMemo } from 'react'
+import {
+  permyriadToPercent,
+  percentToPermyriad,
+  perbicentToPercent,
+  fromWad,
+} from 'utils/format/formatNumber'
+import { decodeFundingCycleMetadata } from 'utils/v1/fundingCycle'
+import { splitPercentFrom, discountRateFrom } from 'utils/v2v3/math'
+
+export const useRelaunchV1ViaV3Create = () => {
+  const router = useRouter()
+
+  const { projectMetadata } = useContext(ProjectMetadataContext)
+  const { queuedFC, queuedPayoutMods, queuedTicketMods } =
+    useContext(V1ProjectContext)
+  const fundingCycleMetadata = decodeFundingCycleMetadata(queuedFC?.metadata)
+
+  const isReady = useMemo(() => {
+    return !(
+      projectMetadata === undefined ||
+      queuedFC === undefined ||
+      queuedPayoutMods === undefined ||
+      queuedTicketMods === undefined
+    )
+  }, [projectMetadata, queuedFC, queuedPayoutMods, queuedTicketMods])
+
+  const v1ConvertedInitialState = useMemo(() => {
+    // No idea why we have to do this way but it is what it is.
+    const convertPayoutMods = (mods: PayoutMod[]) => {
+      const converted: unknown[] = []
+      for (const mod of mods ?? []) {
+        converted.push({
+          percent: splitPercentFrom(
+            parseFloat(permyriadToPercent(mod.percent)),
+          ).toString(),
+          allocator: mod.allocator,
+          beneficiary: mod.beneficiary,
+          lockedUntil: mod.lockedUntil,
+          preferClaimed: mod.preferUnstaked,
+          projectId: !mod.projectId?.eq(0)
+            ? mod.projectId?.toString()
+            : undefined,
+        })
+      }
+      return converted
+    }
+    const convertTicketMods = (mods: TicketMod[]) => {
+      const converted: unknown[] = []
+      for (const mod of mods ?? []) {
```

Comment:
**suggestion** guess we could `map` this instead?

Reply:
map didn't work for me hence why I had to do this :( i'll try the function you suggested above, but this shit is so finicky and old I kind of just wanna keep pushing wdyt

Reply:
ye